### PR TITLE
Add convexity check for multivariate functions

### LINKED ISSUE #26427: Implementing convexity using Hessian Matrix
I saw in #15776 that `is_convex` cannot be used on multivariate functions.

I think it is possible to implement it using `hessian` and `is_positive_semidefinite` (see [this Wikipedia article](https://en.wikipedia.org/wiki/Convex_function)). 

In fact, I gave it a few try and it worked fine, but I think some more testing is required. It is also mandatory to keep the disjunction between univariate and multivariate because `hessian` does not support univariate functions. I'm willing to contribute to the implementation and testing of this feature. However, I'd appreciate some feedback before proceeding further.

**Version** : sympy 1.12

### PR DESCRIPTION
**Pull request by Emilius12**
* calculus
   * Added convexity check for multivariate functions

### CODE REVIEW

File: AUTHORS [hunk 1]
```diff
@@ -1267,3 +1267,4 @@ Han Wei Ang <ang.h.w@u.nus.edu>
 Pablo <48098178+PabloRuizCuevas@users.noreply.github.com>
 Congxu Yang <u7189828@anu.edu.au>
 Saicharan <62512681+saicharan2804@users.noreply.github.com>
+Emile <emile.fourcin1@gmail.com>
```
Comment (oscarbenjamin) [inline AUTHORS:1270]: Don't add yourself to the authors file. That will be done automatically later.

File: .mailmap [hunk 2]
```diff
@@ -548,6 +548,7 @@ Elias Basler <e.e.basler@protonmail.com>
 Elisha Hollander <just4now666666@gmail.com> donno2048 <just4now666666@gmail.com>
 Elliot Marshall <Marshall2389@gmail.com> <marshall2389@gmail.com>
 Elrond der Elbenfuerst <elrond+sympy.org@samba-tng.org>
+Emile <emile.fourcin1@gmail.com>
```
Comment (oscarbenjamin) [inline .mailmap:551]: This should usually look something like:
```
Emile Fourcini <emile.fourcin1@gmail.com> Emile <emile.fourcin1@gmail.com>
```
The first part is the name that should go in the authors file and the second part is the name and email that is recorded in the commits.

File: sympy/calculus/util.py [hunk 3]
```diff
@@ -745,10 +746,13 @@ def is_convex(f, *syms, domain=S.Reals):
     .. [5] https://en.wikipedia.org/wiki/Concave_function
 
     """
-
-    if len(syms) > 1:
-        raise NotImplementedError(
-            "The check for the convexity of multivariate functions is not implemented yet.")
+    if type(*syms) != Symbol :
```
Comment (oscarbenjamin) [inline sympy/calculus/util.py:749]: What does `type(*syms)` do if there is more than one symbol?

File: sympy/calculus/util.py [hunk 4]
```diff
@@ -745,10 +746,13 @@ def is_convex(f, *syms, domain=S.Reals):
...
+        a = hessian(f,*syms)
+        b = a.is_positive_semidefinite
+        if b == None :
+            return None
+        else:
+            return b
```
Comment (oscarbenjamin) [inline sympy/calculus/util.py:755]: This can just be `return a.is_positive_semidefinite`.

File: sympy/calculus/util.py (hunk 3 above)
Comment (Emilius12) [reply sympy/calculus/util.py:749]: I'm going to change that according to [this discussion](https://github.com/sympy/sympy/discussions/26485), it's a mistake from mine due to my error about `hessian`

**Comment by 1e9abhi1e10** [comment]
@Emilius12 Please ensure the pull request title indicates the fix without including issue numbers. Issue numbers belong in the PR description.

File: sympy/calculus/tests/test_util.py [hunk 5]
```diff
@@ -280,18 +280,20 @@ def test_lcim():
...
     assert is_convex(1/x, x, domain=Interval(-oo, 0)) == False
     assert is_convex(x**2, x, domain=Interval(0, oo)) == True
     assert is_convex(1/x**3, x, domain=Interval.Lopen(0, oo)) == True
     assert is_convex(-1/x**3, x, domain=Interval.Ropen(-oo, 0)) == True
-    assert is_convex(log(x), x) == False
-    raises(NotImplementedError, lambda: is_convex(log(x), x, a))
+    assert is_convex(log(x),x) == False
+    assert is_convex(x**2+y**2,x,y)==True
+    assert is_convex(cos(x)+cos(y),x)==False
+    assert is_convex(8*x**2-2*y**2,x,y)==False
```
Comment (smichr) [inline sympy/calculus/tests/test_util.py:292]:
```suggestion
    assert is_convex(log(x) ,x) == False
    assert is_convex(x**2+y**2, x, y) == True
    assert is_convex(cos(x) + cos(y), x) == False
    assert is_convex(8*x**2 - 2*y**2, x, y) == False
```

File: sympy/calculus/util.py [hunk 6]
```diff
@@ -745,18 +746,14 @@ def is_convex(f, *syms, domain=S.Reals):
...
 
     """
-
-    if len(syms) > 1:
-        raise NotImplementedError(
-            "The check for the convexity of multivariate functions is not implemented yet.")
-
+    if len(syms) > 1 :
+        a = hessian(f,syms)
+        return a.is_positive_semidefinite
```
Comment (smichr) [inline sympy/calculus/util.py:751]:
```suggestion
        return hessian(f, syms).is_positive_semidefinite
```

File: .mailmap (hunk 2 above)
Comment (smichr) [inline .mailmap:551]:
```suggestion
Emile Fourcini <emile.fourcin1@gmail.com> Emile <emile.fourcin1@gmail.com>
```

**Pull request merged**